### PR TITLE
created test & fixed array splicing bug

### PR DESCRIPTION
**Pull request by mikedane**
for #181

### CODE REVIEW

File: lib/observable-component.js [hunk 1]
```diff
@@ -92,7 +92,12 @@ function updateObservableWithProps(observable, newProps, observer) {
 			}
 
 			if (type === 'splice') {
-				canReflect.splice(canKey.get(observable, key), values.index, values.deleteCount, values.insert);
+				if (Array.isArray(observable[key])) {
```
Comment (christopherjbaker) [inline lib/observable-component.js]: Should probably use canKey.get here like the next line does, but cache the value so we don't have to look twice.

File: test/connect.js [hunk 2]
```diff
@@ -417,4 +417,40 @@ QUnit.module('@connect with ObserveObject', () => {
 		assert.equal(getTextFromElement(parentDiv), 'abAcdBeCfghi', 'abAcdBeCfghi');
 
 	});
+
+	QUnit.asyncTest('should only use splice on arrays passed into children not objects', (assert) => {
```
Comment (BigAB) [inline test/connect.js]: This test is hard to grok, and the title of the test seems to be describing an implementation detail of the test, not the result we're hoping for the test.

What is breaking, what should it do? Our test title should be what it should do.

File: test/connect.js [hunk 3]
```diff
@@ -417,4 +417,40 @@ QUnit.module('@connect with ObserveObject', () => {
...
+		}
+
+		class ParentStore extends ObserveObject {
+			constructor(props) {
+				super(props);
+				setTimeout((value) => {
+					this.fooObject = new FooList([{ foo0: 'foo0' }], 'bam');
+					this.fooArray = makeFooArray('bam');
+					assert.equal(getTextFromElement(parentDiv), 'bambar', 'bambar');
+					QUnit.start();
```
Comment (BigAB) [inline test/connect.js]: We probably shouldn't use these old, no-longer-documented, Qunit APIs (`QUnit.asyncTest`, `QUnit.start()`, ...etc)  in our tests.
There are new and documented QUnit methods to do these kinds of async tests
https://api.qunitjs.com/assert/async

File: test/connect.js (hunk 2 above)
Comment (BigAB) [reply test/connect.js]: Maybe something like, `should preserve non-numeric key properties on observable arrays passed as props` or something

**Pull request merged**